src/abstractdataframe/reshape.jl

```diff
@@ -398,13 +421,15 @@ function _unstack(df::AbstractDataFrame, rowkeys::AbstractVector{Int},
     Ncol = length(g_colkey)
     col_group_row_idxs = find_group_row(g_colkey)
     colref_map = df[col_group_row_idxs, colkey]
-
     if any(ismissing, colref_map) && !allowmissing
         throw(ArgumentError("Missing value in variable :$(_names(df)[colkey]). " *
                             "Pass `allowmissing=true` to skip missings."))
     end
+    unstacked_val = [fill!(similar(valuecol,
+                                   promote_type(eltype(valuecol), typeof(fillvalue)),
+                                   Nrow),
+                           fillvalue) for _ in 1:Ncol]
```

Comment:
@nalimilan - this definition is problematic for categorical columns the problem is that:
```
julia> using CategoricalArrays

julia> valuecol = categorical(["a"])
1-element CategoricalArray{String,1,UInt32}:
 "a"

julia> fillvalue = ""
""

julia> Nrow = 1
1

julia> similar(valuecol, promote_type(eltype(valuecol), typeof(fillvalue)), Nrow)
1-element Vector{String}:
 #undef

julia> promote_type(eltype(valuecol), typeof(fillvalue))
String
```

Is there a generic idiom that would (without introducing dependency on CategoricalArrays.jl) allow us to take the union of types and if it is OK (like it should be in the example above) produce `CategoricalArray`? Or maybe you think we should change the definition of `similar` in CategoricalArrays.jl?

Reply:
OK - I have thought about it. Given we currently allow:
```
julia> x = categorical([1,2,3])
3-element CategoricalArray{Int64,1,UInt32}:
 1
 2
 3

julia> CategoricalValue(3, x)
CategoricalValue{Int64, UInt32} 3
```
the behavior we have now should be OK.

Reply:
I think it's OK. If we wanted to change this we could adapt `promote_type` in CategoricalArrays, but it's orthogonal to this PR. There would certainly be advantages in promoting to `CategoricalValue`, though it could also create problems (e.g. when concatenating `Vector{Int} and `CategoricalArray{Union{String, Missing}}` the resulting type would be `Any` but CategoricalArrays doesn't handle that well).

Reply:
Yes - I think there would be more problems than benefits. We only need an easy way to create a "stand alone" `CategoricalValue`, as now it is `CategoricalValue(scalar, categorical([scalar]))` which is quite inconvenient.